### ISSUE TITLE
"Literal[...]" is not assignable to "Number" (`number.Number`)

### ISSUE THREAD

**Issue by turbotimon** [microsoft/pyright#11465]
## Environment data

-   Pylance version: 2026.2.1
-   OS and version: Linux kub 6.17.0-22-generic 
-   Python version (& distribution if applicable, e.g. Anaconda): 3.14.3

## Code Snippet

```python
from numbers import Number

def f(x: Number) -> Number:
    return x

assert isinstance(20, Number)

f(20)
```

## Repro Steps

1. Py file with code above in VSCode

## Expected behavior

No error because `20` is a Number

## Actual behavior

```sh
Argument of type "Literal[20]" cannot be assigned to parameter "x" of type "Number" in function "f"
  "Literal[20]" is not assignable to "Number"Pylance[reportArgumentType](https://github.com/microsoft/pylance-release/blob/main/docs/diagnostics/reportArgumentType.md)
```

<img width="908" height="175" alt="Image" src="https://github.com/user-attachments/assets/ddf75260-5ae2-485f-8339-601436a5d17a" />

## Logs

```
2026-05-26 12:28:38.661 [info] (557328): BG(python_pml): indexing: file:///data/home/xxx/_test.py [found 1] 
```

**Comment by rchiodo** [comment]
I find this strange but it seems this is by design. You can't assign an int to a Number:

https://github.com/microsoft/pyright/issues/11093

**Comment by rchiodo** [comment]
This issue as well goes into more detail:
https://github.com/microsoft/pyright/issues/4616

**Comment by rchiodo** [comment]
Pyrefly gives a similar error:

<img width="1033" height="133" alt="Image" src="https://github.com/user-attachments/assets/e2fe8a87-839a-4098-a4cb-91686ee99464" />

**Comment by turbotimon** [comment]
Oh Python, what have you done..

Especially since `isinstance(20, Number) == True` as well as `issubclass(int, numbers.Number) == True` and at the same time [*"The Number class and int class are both nominal types with no overlap"*](https://github.com/microsoft/pyright/issues/11093#issuecomment-3489116292) 😵‍💫

Anyway, thanks for the clarification! (Altough I still do not understand it, but can't blame Pyright for that)
